### PR TITLE
Feature/store cluster config

### PR DESCRIPTION
**Pull request by luigibarbato**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -27,6 +27,7 @@ require (
 	github.com/sighupio/fury-distribution v1.24.1-0.20230103153037-ad0c881984b3
 	golang.org/x/exp v0.0.0-20221217163422-3c43f8badb15
 	gopkg.in/yaml.v3 v3.0.1
+	k8s.io/client-go v0.26.0
```
Comment (Al-Pragliola) [inline go.mod:30]: is client-go still necessary?

File: internal/apis/kfd/v1alpha2/eks/create/kubernetes.go [hunk 2]
```diff
@@ -106,11 +107,15 @@ func (k *Kubernetes) Exec() error {
 	}
 
 	if err := k.tfRunner.Init(); err != nil {
-		return fmt.Errorf("error running terraform init: %w", err)
+		logrus.Debugf("error running terraform init: %s", err)
+
+		return errPhase
```
Comment (Al-Pragliola) [inline internal/apis/kfd/v1alpha2/eks/create/kubernetes.go:112]: the error is already wrapped in the creator with a similar message: "an error occurred while running kubernetes phase", if we return errPhase here we repeat two times the same message

File: internal/x/kube/config.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+	err = os.WriteFile(path.Join(currentDir, "kubeconfig"), kubeconfig, iox.FullRWPermAccess)
+	if err != nil {
+		return fmt.Errorf("error writing kubeconfig file: %w", err)
+	}
+
+	return nil
+}
+
+func GetConfigFromFile(kubeConfigPath string) (*rest.Config, error) {
```
Comment (Al-Pragliola) [inline internal/x/kube/config.go:66]: is this function still needed?

File: internal/x/kube/config.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	"path/filepath"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	iox "github.com/sighupio/furyctl/internal/x/io"
+)
+
+func CreateConfig(data []byte, p string) (string, error) {
+	err := os.WriteFile(path.Join(p, "kubeconfig"), data, iox.FullRWPermAccess)
```
Comment (Al-Pragliola) [inline internal/x/kube/config.go:20]: you can refactor with newly created iox.WriteFile

File: internal/x/kube/config.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	if err != nil {
+		return fmt.Errorf("error getting kubeconfig absolute path: %w", err)
+	}
+
+	kubeconfig, err := os.ReadFile(kubePath)
+	if err != nil {
+		return fmt.Errorf("error reading kubeconfig file: %w", err)
+	}
+
+	err = os.WriteFile(path.Join(currentDir, "kubeconfig"), kubeconfig, iox.FullRWPermAccess)
```
Comment (Al-Pragliola) [inline internal/x/kube/config.go:58]: you can refactor with newly created iox.WriteFile

**Review by Al-Pragliola — CHANGES_REQUESTED** [review]

**Review by Al-Pragliola — COMMENTED** [review]
the pipeline is failing on e2e tests

File: internal/apis/kfd/v1alpha2/eks/create/kubernetes.go (hunk 2 above)
Comment (luigibarbato) [reply internal/apis/kfd/v1alpha2/eks/create/kubernetes.go:112]: the linter fails if we don't wrap the error

**Review by Al-Pragliola — APPROVED** [review]
LGTM

**Review by omissis — APPROVED** [review]

**Pull request merged**